napari/_check_numpy_version.py

```diff
@@ -0,0 +1,73 @@
+"""
+This module is used to prevent a known issue with numpy<2 on macOS arm64
+architecture installed from pypi wheels.
+Setting thread limits is inspired on threadpoolctl package to prevent adding
+the dependency to napari.
+However, if there is some problem in future it should be ok to
+add threadpoolctl as a dependency to napari and use it directly.
+"""
+
+import ctypes
+import logging
+import os
+import platform
+from pathlib import Path
+
+import numpy as np
+from packaging.version import parse as parse_version
+
+# if run with numpy<2 on macOS arm64 architecture compiled from pypi wheels,
+# then it will crash with bus error if numpy is used in different thread
+# Issue reported https://github.com/numpy/numpy/issues/21799
+if (
+    parse_version(np.__version__) < parse_version('2')
+    and platform.system() == 'Darwin'
+    and platform.machine() == 'arm64'
+):  # pragma: no cover
+    try:
+        PROBLEMATIC_NUMPY_MACOS = (
+            'cibw-run' in np.show_config('dicts')['Python Information']['path']  # type: ignore
```

Comment:
As the issue is specific to OpenBLAS is it worth also checking for that?
something like `np.show_config(mode='dicts')['Build Dependencies']['blas']['name'] == 'openblas64'`
Then again is a `cibw-run` wheel but without openblas likely/possible?

Reply:
And are we sure this threading thing is arm64 only? Or does x86 numpy use the intel blas...

Reply:
> something like `np.show_config(mode='dicts')['Build Dependencies']['blas']['name'] == 'openblas64'`
> Then again is a `cibw-run` wheel but without openblas likely/possible?

This is an issue with a specific version of openblas used by numpy when compiling wheels. So decided to check build information. As someone may have own compilation against more modern openblas

> And are we sure this threading thing is arm64 only? Or does x86 numpy use the intel blas...

On CI, only the arm64 version crashes.